mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the segmentation.
+        If none is provided, a new set of axes is created.
+
+    kwargs
+        Additional keyword arguments to pass to
+        `matplotlib.patches.Rectangle`.
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes._subplots.AxesSubplot
+        A handle to the (possibly constructed) plot axes
+    '''
+    if text_kw is None:
+        text_kw = dict(va='top',
+                       clip_on=True,
+                       bbox=dict(boxstyle='round', facecolor='white'))
+
+    seg_def_style = dict(linewidth=1)
+
+    if ax is None:
+        # Create a new axis
+        ax = plt.gca()
+        ax.set_ylim([0, 1])
+
+    # Infer height
+    if base is None:
+        base = ax.get_ylim()[0]
+
+    if height is None:
+        height = ax.get_ylim()[1]
+
+    cycler = ax._get_patches_for_fill.prop_cycler
+
+    seg_map = dict()
+
+    for lab in labels:
+        if lab in seg_map:
+            continue
+
+        style = next(cycler)
+        seg_map[lab] = seg_def_style.copy()
+        seg_map[lab].update(style)
+        seg_map[lab]['facecolor'] = seg_map[lab].pop('color')
+        seg_map[lab].update(kwargs)
+        seg_map[lab]['label'] = lab
+
+    for ival, lab in zip(intervals, labels):
+        rect_kwargs = seg_map[lab]
+        rect = Rectangle((ival[0], base), ival[1] - ival[0], height,
+                         **rect_kwargs)
+        ax.add_patch(rect)
+        rect_kwargs.pop('label', None)
+
+        if text:
+            ann = ax.annotate(lab,
+                              xy=(ival[0], height), xycoords='data',
+                              xytext=(8, -10), textcoords='offset points',
+                              **text_kw)
+            ann.set_clip_path(rect)
+
+    ax.set_yticks([])
+
+    __expand_limits(ax, [intervals.min(), intervals.max()], which='x')
+
+    return ax
+
+
+def labeled_intervals(intervals, labels, label_set=None,
+                      base=None, height=None, extend_labels=True,
+                      ax=None, tick=True, **kwargs):
+    '''Plot labeled intervals with each label on its own row.
+
+    Parameters
+    ----------
+    intervals : np.ndarray, shape=(n, 2)
+        segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    labels : list, shape=(n,)
+        reference segment labels, in the format returned by
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    label_set : list
+        An (ordered) list of labels to determine the plotting order.
+        If not provided, the labels will be inferred from existing
+        `yticklabels`.
+        If no `yticklabels` exist, then the sorted set of unique values
+        in `labels` is taken as the label set.
+
+    base : np.ndarray, shape=(n,), optional
+        Vertical positions of each label
+
+    height : np.ndarray, shape=(n,), optional
```

Comment:
Huh, I guess this one got buried under "outdated diff" while I was working through it.

Yes, I can add this.